middle-pgsql.cpp

```diff
@@ -323,41 +323,41 @@ int middle_pgsql_t::local_nodes_set(const osmid_t& id, const double& lat,
       char *buffer = (char *)alloca( length );
 #ifdef FIXED_POINT
       ramNode n(lon, lat);
-      if( snprintf(buffer, length, "%" PRIdOSMID "\t%d\t%d\t%s\n",
-                   id, n.int_lat(), n.int_lon(), tag_buf ) > (length-10) )
-      { fprintf( stderr, "buffer overflow node id %" PRIdOSMID "\n", id); return 1; }
+      if (snprintf(buffer, length, "%" PRIdOSMID "\t%d\t%d\t%s\n",
+                   id, n.int_lat(), n.int_lon(), tag_buf ) > (length-10)) {
+         throw std::runtime_error((boost::format("Buffer overflow node id %1%") % id).str());
```

Comment:
This is a change from continuing to execute to exiting the program, but I've never seen this error occur.

I also thing the entire section should be rewritten to not C strings, but that should be a different PR.